File: src/main/java/cubesrv/ProgressBase.java
package cubesrv;

import java.util.concurrent.locks.ReentrantLock;

public class ProgressBase {
    private static final ReentrantLock gProgressLock = new ReentrantLock();
    private static volatile boolean m_isStopRequested = false;

    public static void mutexLock() {
        gProgressLock.lock();
    }

    public static void mutexUnlock() {
        gProgressLock.unlock();
    }

    public static boolean isStopRequested() {
        return m_isStopRequested;
    }

    public static void requestStop() {
        mutexLock();
        m_isStopRequested = true;
        mutexUnlock();
    }

    public static void requestRestart() {
        mutexLock();
        m_isStopRequested = false;
        mutexUnlock();
    }
}

File: src/main/kotlin/cubesrv/ProgressBase.kt
package cubesrv

import kotlinx.coroutines.sync.Mutex

open class ProgressBase {
    companion object {
        val gProgressMutex = Mutex()
        var m_isStopRequested = false

        suspend fun mutexLock() : Unit {
            gProgressMutex.lock()
        }

        fun isStopRequested() : Boolean = m_isStopRequested

        fun mutexUnlock() : Unit {
            gProgressMutex.unlock()
        }

        suspend fun requestStop() : Unit
        {
            mutexLock()
            m_isStopRequested = true
            mutexUnlock()
        }

        suspend fun requestRestart() : Unit
        {
            mutexLock()
            m_isStopRequested = false
            mutexUnlock()
        }
    }
}

